add ScriptUsing*StringResource tests

mykmelez opened the pull request:
This branch adapts two of the ExternalStringResource tests in test-api.cc to tests in value.cc. Most of the test assertions are commented out, because SpiderShim doesn't yet support the features being tested (or the test harness doesn't support them), like String::IsExternal and String::GetExternalStringResourceBase.

So I'm not sure how useful these tests are, nor how useful it would be to adapt more of the ExternalStringResource tests in test-api.cc. Of course we could always implement those features, but since Node doesn't depend on them, they don't seem like a high priority.

ehsan commented:
> So I'm not sure how useful these tests are, nor how useful it would be to adapt more of the ExternalStringResource tests in test-api.cc. Of course we could always implement those features, but since Node doesn't depend on them, they don't seem like a high priority.

Agreed!

ehsan commented on deps/spidershim/test/v8engine.h:
```diff
@@ -95,6 +95,14 @@ class V8Engine {
     return Local<Value>();
   }
 
+  Local<Value> CompileRun(Local<Context> context, Local<String>* script) {
```
Passing `Local` by pointer doesn't make sense.  Did you mean to pass it by value?

Also it would be nice if you refactored the other `CompileRun` overload to be implemented in terms of this using `v8_str` or something like that...

mykmelez replied:
> Also it would be nice if you refactored the other CompileRun overload to be implemented in terms of this using v8_str or something like that...

Yup, makes sense, done in eaa88ed.

ehsan commented:
Looks good except for the one comment I posted on `CompileRun`...

mykmelez commented:
> Looks good except for the one comment I posted on CompileRun...

Fixed in 5209375. In the process, I realized that the _context_ parameter is unused, so I removed it in d37915d. (That's also true for the CompileRun variant that takes a `const char* script`, which originally used that parameter but doesn't anymore.)
